### DISCUSSION TITLE
Which Event is fired when Bot Leave/Join

### DISCUSSION THREAD

**Discussion by heatdream** [bwmarrin/discordgo#1294] [Q&A]
Hey People's,
I want to trigger an function when the bot leaves or joins a guild. The question is now: Which event is fired, so that i can fire up my funktions?

**Comment by Nv7-GitHub** [comment]
`GuildCreate` is fired when bot joins server

**Comment by FedorLap2006** [comment]
You can listen for `*discordgo.GuildCreate` (as @Nv7-GitHub already mentioned) to detect whether bot joins the guild. But beware, it is also fired on startup, when bot receives initial guild data.

As for leaving, you would want `*discordgo.GuildDelete`. However it also comes with an asterisk. It may fire when guild becomes `Unavailable` (in maintanence/outage), so be sure to check that this field is false.
